### DISCUSSION TITLE
Server Function Endpoint

### DISCUSSION THREAD

**Discussion by Tibuarka** [leptos-rs/leptos#2447] [Q&A]
I stumbled upon the fact that my registered server functions generate an endpoint which does not reflect the endpoint I declared. For example '/api/ping' turns into '/api/ping/ping1234567890'. Is this from leptos or from the server_fn crate? And if it is by design from leptos why is that so and is there a way to change the generated endpoint url?

**Comment by gbj** [comment]
To be clear are you specifying an endpoint using the argument in server functions intended to do so? What would be some example code?
```rust
#[server(
  name = SomeStructName,
  prefix = "/my_api",
  endpoint = "my_fn",
```
If you are using the above and it's still generating a hash at the end it's a bug. If you aren't specifying `endpoint = ` then you can do that to get a permanent, set endpoint.

**Reply by Tibuarka** [reply]
Ah yes, my stupidity has struck again because who needs to read instructions...
Yeah, I only specified the prefix part because I used axum alot lately and this behaviour must have carried over.
Now to another thing, can I specifiy the allowed method for server functions? If I understand correctly the default method allowed is 'POST'?

**Reply by gbj** [reply]
The various server function input encodings support various methods, linked to the different encodings for their arguments. `GetUrl` supports a `GET` request with the arguments encoded in the URL query string, the other encodings support `POST` requests with the arguments in the body. There is not built-in support for other methods, although they can be created as custom encodings (see the `server_fns_axum` example in the repo for an example of creating custom encodings).

(Note that `GET` and `POST` are the only supported methods for HTML forms, so they are usually preferable for graceful degradation if WASM is not available, if you are using server functions with forms)

**Reply by Tibuarka** [reply]
Alright a little bit confusing because the encoding parameter of the server macro is deprecated/not working. I set the input parameter to 'GetUrl' which seems to be working. Now everytime I try to access the endpoint directly it redirects me to my leptos homepage. I guess this is correct?

**Reply by gbj** [reply]
Sorry, I'm not able to follow along with what you're describing. Feel free to share example code.
